src/main/java/com/g4s8/ghman/web/GHAuthorizationTake.java

```diff
@@ -0,0 +1,62 @@
+package com.g4s8.ghman.web;
+
+import java.io.IOException;
+import javax.sql.DataSource;
+import com.g4s8.ghman.user.User;
+import com.g4s8.ghman.user.Users;
+import com.jcabi.http.request.JdkRequest;
+import com.jcabi.http.response.JsonResponse;
+import com.jcabi.log.Logger;
+import com.jcabi.http.Request;
+import org.takes.Response;
+import org.takes.Take;
+import org.takes.facets.auth.RqAuth;
+import org.takes.rq.RqHref;
+import org.takes.rs.RsText;
+
+/**
+ * Github Authorization Take.
+ * @since 1.0
+ * @checkstyle LineLengthCheck (500 lines)
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ * @checkstyle ClassFanOutComplexityCheck (500 lines)
+ */
+final class GHAuthorizationTake implements Take {
+
+    private DataSource data;
+
+    /**
+     * Ctor.
+     * @param data Data source
+     */
+    public GHAuthorizationTake(final DataSource data) {
+        this.data = data;
+    }
+
+    @Override
+    public final Response act(final org.takes.Request req) throws IOException {
+        final String code = new RqHref.Smart(req)
+            .single("code");
+        final String token = new JdkRequest("https://github.com/login/oauth/access_token")
+            .method(Request.POST)
+            .uri()
+            .queryParam("client_id", System.getenv("GH_CLIENT"))
+            .queryParam("client_secret", System.getenv("GH_SECRET"))
+            .queryParam("code", code).back()
+            .header("Accept", "application/json")
+            .fetch()
+            .as(JsonResponse.class)
+            .json()
+            .readObject()
+            .getString("access_token");
+        final String urn = new RqAuth(req).identity().urn();
+        final User user = new Users(data).user(Long.parseLong(urn.split(":")[2]));
+        user.authorize(token);
+        Logger.info(
```

Comment:
@samihibrahim please a todo to replace those static loggers with something more OO (maybe related with https://www.yegor256.com/2019/03/19/logging-without-static-logger.html)